### PR TITLE
Feature/er 1236 add material selection form

### PR DESCRIPTION
**Pull request by martinyde**
https://jira.itkdev.dk/browse/ER-1236

- Add form for material selection
- Add admin form
- Modify code to respect multiple forms

billede

billede

### CODE REVIEW

File: ereolen_support/ereolen_material.form.inc [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+ *
+ * @return mixed
+ *
+ * @throws \Exception
+ */
+function ereolen_material_form($form, &$form_state) {
+  $form['#attributes'] = [
+    'name' => 'ereolen_material_form',
+    'class' => ['node-webform'],
+    'onsubmit' => 'op.disabled = true; return true;'
```
Comment (rimi-itk) [inline ereolen_support/ereolen_material.form.inc]: Is this a known Drupal trick? Is `op` defined globally?

File: ereolen_support/ereolen_material.form.inc [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  ];
+
+  $form['#attached']['css'] = [
+    drupal_get_path('module', 'ereolen_support') . '/css/styles.css'
+  ];
+
+  $form['ereolen_material_lead'] = [
+    '#markup' => theme('ereolen_support_lead_text', [
+      'text' =>  variable_get('ereolen_material_lead')['value'] ?? NULL,
```
Comment (rimi-itk) [inline ereolen_support/ereolen_material.form.inc:34]: 🎉

File: ereolen_support/ereolen_material.form.inc [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    '#title' => t('Description'),
+    '#required' => TRUE,
+  ];
+
+  $form['submit_button'] = [
+    '#type' => 'submit',
+    '#value' => t('Send'),
+  ];
+
+  honeypot_add_form_protection($form, $form_state, array('honeypot', 'time_restriction'));
```
Comment (rimi-itk) [inline ereolen_support/ereolen_material.form.inc]: Use short array notation here (`[…]`) for consistency with the rest of your code.

File: ereolen_support/ereolen_support.form.inc [hunk 4]
```diff
@@ -455,9 +455,9 @@ function ereolen_support_form_submit($form, &$form_state) {
   }
   else {
     // Check for and create user in Jira
-    $jira_user = _ereolen_support_search_jira_user($user, $form_state);
+    $jira_user = _ereolen_support_search_jira_user($user, $form_state, $form_state['values']['ereolen_support_email']);
```
Comment (rimi-itk) [inline ereolen_support/ereolen_support.form.inc:458]: Consider storing `$form_state['values']['ereolen_support_email']` in a local variable.

File: ereolen_support/ereolen_support.helper.inc [hunk 5]
```diff
@@ -17,10 +17,9 @@
  *
  * @throws \Exception
  */
-function _ereolen_support_search_jira_user($user, $form_state) {
-  $form_email = $form_state['values']['ereolen_support_email'];
+function _ereolen_support_search_jira_user($user, $form_state, $email) {
   $requestParts = _ereolen_support_get_request_parts('GET');
-  $result = drupal_http_request($requestParts['jira_path'] . '/user/search?includeInactive=true&username=' . $form_email, $requestParts['options']);
+  $result = drupal_http_request($requestParts['jira_path'] . '/user/search?includeInactive=true&username=' . $email, $requestParts['options']);
```
Comment (rimi-itk) [inline ereolen_support/ereolen_support.helper.inc:22]: You may want to use https://www.php.net/manual/en/function.urlencode.php here. Or https://www.php.net/manual/en/function.http-build-query.php.

File: ereolen_support/ereolen_support.helper.inc [hunk 6]
```diff
@@ -192,6 +190,34 @@ function _ereolen_support_send_to_jira($form, &$form_state, $jiraUser) {
...
+function _ereolen_material_send_to_jira($form, &$form_state, $jiraUser) {
+  $data = [
+    'fields' => [
+      'project' => [
+        'id' => variable_get('jira_material_project_id'),
+      ],
+      'issuetype' => [
+        'id' => variable_get('jira_material_issue_type_id'),
+      ],
+      'summary' => preg_replace('/\s+/', ' ', check_plain(trim(substr($form_state['values']['ereolen_material_description'], 0, 30)))),
```
Comment (rimi-itk) [inline ereolen_support/ereolen_support.helper.inc:202]: Maybe you could use https://api.drupal.org/api/drupal/includes!unicode.inc/function/truncate_utf8/7.x here for the actual trimming, but the code here is also fine.

**Review by rimi-itk — CHANGES_REQUESTED** [review]
Looks good. Some minor changes needed.

File: ereolen_support/ereolen_material.form.inc (hunk 1 above)
Comment (martinyde) [reply ereolen_support/ereolen_material.form.inc]: I'm actually not sure where it's from it's a copy from ereolen_support_form but where that is fetched from i don't know. However with honeypot installed i don't think we need it anymore.

File: ereolen_support/ereolen_support.form.inc (hunk 4 above)
Comment (martinyde) [reply ereolen_support/ereolen_support.form.inc:458]: Noted!

File: ereolen_support/ereolen_support.helper.inc (hunk 5 above)
Comment (martinyde) [reply ereolen_support/ereolen_support.helper.inc:22]: Noted!

File: ereolen_support/ereolen_support.helper.inc (hunk 6 above)
Comment (martinyde) [reply ereolen_support/ereolen_support.helper.inc:202]: Noted!

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**